process_metrics_block.py

```diff
@@ -0,0 +1,104 @@
+import psutil
+
+from nio.common.block.base import Block
+from nio.common.signal.base import Signal
+from nio.common.command import command
+from nio.common.discovery import Discoverable, DiscoverableType
+from nio.metadata.properties import ObjectProperty, ExpressionProperty, \
+    PropertyHolder, BoolProperty
+from nio.modules.scheduler import Job
+
+
+RETRY_LIMIT = 3
+
+
+class Menu(PropertyHolder):
+    
+    virtual_memory = BoolProperty(title='Virtual Memory', default=True)
+    memory_percent = BoolProperty(title='Memory Percentage', default=True)
+    cpu_percent = BoolProperty(title='CPU Percentage', default=True)
+    num_ctx_switches = BoolProperty(title='Number of Context Switches', default=True)
+    num_fds = BoolProperty(title='Number of File Descriptors', default=True)
+    is_running = BoolProperty(title='Is Running?', default=True)
+    
+    children = BoolProperty(title='Children')
+    threads = BoolProperty(title='Threads')
+    cmd_line = BoolProperty(title='Command Line')
+
+
+class NoPIDException(Exception):
+    pass
+
+
+@Discoverable(DiscoverableType.block)
+class ProcessMetrics(Block):
+
+    menu = ObjectProperty(Menu, title='Menu')
+    pid_expr = ExpressionProperty(title='PID Expression',
+                                  allow_none=True,
+                                  attr_default=NoPIDException)
+
+    def __init__(self):
+        super().__init__()
+        self._retry_count = 0
+
+    def process_signals(self, signals):
+        for sig in signals:
+            try:
+                pid = self.pid_expr(sig)
```

Comment:
If I remember correctly, expression properties can raise exceptions if the expression can't evaluate so these always need to be wrapped in a try (which you do) with an except clause to handle Exception.